src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Repositories/OrganisationRepository.cs

```diff
@@ -17,101 +17,113 @@ namespace Altinn.Platform.Authentication.Repositories
+                .SetAbsoluteExpiration(new TimeSpan(1, 0, 0));
         }
 
         /// <inheritdoc/>
         public async Task<Organisation> GetOrganisationByOrgNumber(string orgNumber)
         {
-            await HarvestOrgsIfCacheIsMoreThanOneHourOld();
+            string cacheKey = $"org-{orgNumber}";
+            Organisation organisation;
 
-            return orgNumberToOrganisation.GetValueOrDefault(orgNumber, null);
+            if (!_memoryCache.TryGetValue(cacheKey, out organisation))
+            {
+                await HarvestOrgs();
+                _memoryCache.TryGetValue(cacheKey, out organisation);
+            }
+
+            return organisation;
         }
 
         /// <inheritdoc/>
         public async Task<Organisation> GetOrganisationByOrg(string org)
         {
-            await HarvestOrgsIfCacheIsMoreThanOneHourOld();
+            string cacheKey = $"org-{org}";
+            Organisation organisation;
+
+            if (!_memoryCache.TryGetValue(cacheKey, out organisation))
+            {
+                await HarvestOrgs();
+                _memoryCache.TryGetValue(cacheKey, out organisation);
+            }
 
-            return orgToOrganisation.GetValueOrDefault(org, null);
+            return organisation;
         }
 
         /// <inheritdoc/>
         public async Task<string> LookupOrg(string orgNumber)
         {
-            await HarvestOrgsIfCacheIsMoreThanOneHourOld();
+            string cacheKey = $"org-{orgNumber}";
+            Organisation organisation;
 
-            Organisation organisation = orgNumberToOrganisation.GetValueOrDefault(orgNumber, null);
+            if (!_memoryCache.TryGetValue(cacheKey, out organisation))
+            {
+                await HarvestOrgs();
+                _memoryCache.TryGetValue(cacheKey, out organisation);
+            }
 
             return organisation?.Org;
         }
 
         /// <inheritdoc/>
         public async Task<string> LookupOrgNumber(string org)
         {
-            await HarvestOrgsIfCacheIsMoreThanOneHourOld();
-
-            Organisation organisation = orgToOrganisation.GetValueOrDefault(org, null);
-
-            return organisation?.OrgNumber;
-        }
-
-        private async Task HarvestOrgsIfCacheIsMoreThanOneHourOld()
-        {
-            DateTime timestamp = DateTime.Now;
-            timestamp = timestamp.AddHours(-1);
+            string cacheKey = $"org-{org}";
+            Organisation organisation;
 
-            if (dictionaryLastUpdated < timestamp || orgNumberToOrganisation.Count == 0)
+            if (!_memoryCache.TryGetValue(cacheKey, out organisation))
             {
-               await HarvestOrgs();
+                await HarvestOrgs();
+                _memoryCache.TryGetValue(cacheKey, out organisation);
             }
+
+            return organisation?.OrgNumber;
         }
 
         /// <inheritdoc/>
         public async Task HarvestOrgs()
```

Comment:
This method should probably not be a part of `IOrganisationRepository`. Unless we want users of the interface the ability to call it.